crankshaft/compiler.md

```diff
@@ -0,0 +1,281 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](http://doctoc.herokuapp.com/)*
+
+- [v8 Compiler](#v8-compiler)
+  - [Components](#components)
+    - [Base Compiler](#base-compiler)
+    - [Runtime Profiler](#runtime-profiler)
+    - [Optimizing Compiler](#optimizing-compiler)
+    - [Deoptimization Support](#deoptimization-support)
+  - [Optimized Code vs. Inline Caches and Unoptimized Code](#optimized-code-vs-inline-caches-and-unoptimized-code)
+    - [Full Compiler](#full-compiler)
+    - [Inline Caches](#inline-caches)
+    - [Monomorphism vs. Polymorphism](#monomorphism-vs-polymorphism)
+      - [Considerations](#considerations)
+    - [Optimizing Compiler](#optimizing-compiler-1)
+    - [Deoptimization](#deoptimization)
+    - [Causes for Deoptimization](#causes-for-deoptimization)
+      - [Modifying Object Shape](#modifying-object-shape)
+        - [Considerations](#considerations-1)
+  - [Efficiently Representing Values and Tagging](#efficiently-representing-values-and-tagging)
+    - [Considerations](#considerations-2)
+  - [Arrays](#arrays)
+    - [Fast Elements](#fast-elements)
+      - [Characteristics](#characteristics)
+    - [Dictionary Elements](#dictionary-elements)
+      - [Characteristics](#characteristics-1)
+    - [Double Array Unboxing](#double-array-unboxing)
+    - [Typed Arrays](#typed-arrays)
+      - [Float64Array](#float64array)
+    - [Considerations](#considerations-3)
+  - [Resources](#resources)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# v8 Compiler
+
+## Components
+
+### Base Compiler 
```

Comment:
The *Base Compiler* is the Ignition interpreter.

Reply:
Sorry should have made this clear, but all documents inside the `./crankshaft` folder are from the previous version and kept there only in case people want to compare or need to see how things used to work.

I'm mainly interested in reviews of the new content at the root of this repo (basically the ones I linked in the PR message).

However any of your comments that also apply to newer content I will apply there.

Reply:
So in this case that'd be the [compiler.md](https://github.com/thlorenz/v8-perf/blob/turbo-updates/compiler.md)